### PR TITLE
[CLC-165]: Add details to Snapshot List Command Output

### PR DESCRIPTION
**Pull request by kutluhanmetin**

### CODE REVIEW

File: internal/serialization/snapshot.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	}
+	return nil
+}
+
+func (SnapshotFactory) FactoryID() int32 {
+	return snapshotFactoryID
+}
+
+func (s *Snapshot) WriteData(output serialization.DataOutput) {
+
```
Comment (yuce) [inline internal/serialization/snapshot.go:42]:
```suggestion
// not used
```

File: internal/serialization/snapshot.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	return snapshotClassID
+}
+
+type SnapshotFactory struct{}
+
+func (SnapshotFactory) Create(classID int32) serialization.IdentifiedDataSerializable {
+	if classID == snapshotClassID {
+		return &Snapshot{}
+	}
+	return nil
```
Comment (yuce) [inline internal/serialization/snapshot.go:34]: It is better to panic here instead of returning `nil` so the actual error is known.

**Review by yuce — CHANGES_REQUESTED** [review]
Two minor ideas...

File: internal/serialization/snapshot.go (hunk 2 above)
Comment (kutluhanmetin) [reply internal/serialization/snapshot.go]: https://github.com/hazelcast/hazelcast-commandline-client/pull/295/commits/ea6f747894efa10efd50b5296f03673e644117cb

File: internal/serialization/snapshot.go (hunk 1 above)
Comment (kutluhanmetin) [reply internal/serialization/snapshot.go]: https://github.com/hazelcast/hazelcast-commandline-client/pull/295/commits/ea6f747894efa10efd50b5296f03673e644117cb

**Review by yuce — APPROVED** [review]
Looks good!

**Pull request merged**